src/main/java/rx/internal/operators/OperatorSampleWithObservable.java

```diff
@@ -57,16 +59,18 @@ public void onNext(U t) {
 
             @Override
             public void onError(Throwable e) {
+                // no need to null check, main is assigned before any of the two gets subscribed
+                main.get().unsubscribe();
                 s.onError(e);
-                unsubscribe();
             }
 
             @Override
             public void onCompleted() {
+                // no need to null check, main is assigned before any of the two gets subscribed
+                main.get().unsubscribe();
```

Comment:
Why change the order of `unsubscribe` and `onCompleted `? Is it possible that `s.onCompleted` uses some resource in `main` and that will be destroyed in `unsubscribe`?